### PR TITLE
Pr/fix clipboard callbacks

### LINKED ISSUE #1027: xfixes forwarding not working
In current HEAD (b7034724abb859db36a3fa4312e4827141114f45) there's a bug regarding xfixes forwarding for selection changes. I think I know how to fix it but I am lacking time.

The problem is as follows:
- run a desktop nx session ("outer nxagent")
- from that session run an rootless nx session with konsole ("inner nxagent")
- make a selection in the konsole inside the rootless session -> clipboard owner is konsole
- make a selection on the real X server -> the outer nxagent will get a Xfixes SelectionSetOwner event from the real X server but will not trigger that event for the inner nxagent
- if the previous clipboard owner (konsole) relies on getting Xfixes events on every clipboard change and does not try to use XGetSelectionOwner() it will assume it is still the owner. The effect is that pastes in konsole will refer to itself and thus paste the previous clipboard content. This problem can e.g. be seen in konsole from RHEL 7.9 but not in the one from debian 10.

The source of all this is 7aa969cd4ee5fe6ecf74f82442e4a0a7491191c1 where I was driven by wanting to get rid of the trap and totally missed what the callback is required for.

I will re-implement the old behaviour soon (a simple revert is not working due to other changes), maybe I can drop the trap by other means.

### PR DESCRIPTION
**Pull request by uli42**
This fixes #1027 thus removing a release blocker

### CODE REVIEW

**Comment by sunweaver** [comment]
@uli42: Don't forget to add me as reviewer, when ready with this.

**Comment by uli42** [comment]
it _IS_ ready

File: nx-X11/programs/Xserver/hw/nxagent/Clipboard.c [hunk 1]
```diff
@@ -921,15 +929,34 @@ void nxagentHandleSelectionClearFromXServerByIndex(int index)
...
     clearSelectionOwnerData(index);
 
     setClientSelectionStage(index, SelectionStageNone);
 
     invalidateTargetCache(index);
+
+    /*
+     * now call the callbacks. This is important, especially for
+     * xfixes events to work properly. Keep in mind that this will
+     * also call out own callback so we must be prepared there to not
```
Comment (sunweaver) [inline nx-X11/programs/Xserver/hw/nxagent/Clipboard.c]: s/own callback/own callbacks/ ?

Comment (sunweaver) [reply nx-X11/programs/Xserver/hw/nxagent/Clipboard.c]: or s/out/our/ ?

File: nx-X11/programs/Xserver/hw/nxagent/Clipboard.c [hunk 2]
```diff
@@ -95,7 +95,7 @@ const int nxagentMaxSelections = 2;
 
 /* store the remote atom for all selections */
 static XlibAtom *remoteSelectionAtoms = NULL;
-static Atom *localSelelectionAtoms = NULL;
+static Atom *localSelectionAtoms = NULL;
```
Comment (sunweaver) [inline nx-X11/programs/Xserver/hw/nxagent/Clipboard.c:98]: Did I miss that? I obviously did. Tsts...

File: nx-X11/programs/Xserver/hw/nxagent/Clipboard.c [hunk 3]
```diff
@@ -2394,16 +2394,7 @@ static void setSelectionOwnerOnXServer(Selection *pSelection)
...
-              nxagentClientInfoString(lastSelectionOwner[index].client),
-                  nxagentClientInfoString(pSelection->client));
-  fprintf(stderr, "%s: lastSelectionOwner.window [0x%x] -> [0x%x]\n", __func__,
-              lastSelectionOwner[index].window, pSelection->window);
-  fprintf(stderr, "%s: lastSelectionOwner.windowPtr [%p] -> [%p] [0x%x] (serverWindow: [0x%lx])\n", __func__,
-              (void *)lastSelectionOwner[index].windowPtr, (void *)pSelection->pWin,
-                  pSelection->pWin ? nxagentWindow(pSelection->pWin) : 0, serverWindow);
-  fprintf(stderr, "%s: lastSelectionOwner.lastTimeChanged [%u]\n", __func__,
-              lastSelectionOwner[index].lastTimeChanged);
+  printSelectionStat(index);
```
Comment (sunweaver) [inline nx-X11/programs/Xserver/hw/nxagent/Clipboard.c:2398]: Shouldn't the part above be shipped in a separate commit?

Comment (sunweaver) [reply nx-X11/programs/Xserver/hw/nxagent/Clipboard.c:2398]: This doesn't feel related to the TimeStamp data type change...

File: nx-X11/programs/Xserver/hw/nxagent/Clipboard.c [hunk 4]
```diff
@@ -877,8 +877,9 @@ void invalidateTargetCaches(void)
 
 /*
  * This is called from Events.c dispatch loop on reception of a
- * SelectionClear event. We receive this event if someone on the real
- * X server claims the selection ownership we have/had.
+ * SelectionClear or a XFixes selection event from the real X
```
Comment (sunweaver) [inline nx-X11/programs/Xserver/hw/nxagent/Clipboard.c]: s/a Xfixes/an Xfixes/

**Review by sunweaver — CHANGES_REQUESTED** [review]
Some minor changes for the comments. Please amend and then let's merge.

File: nx-X11/programs/Xserver/hw/nxagent/Clipboard.c (hunk 1 above)
Comment (uli42) [reply nx-X11/programs/Xserver/hw/nxagent/Clipboard.c]: no, we only have one callback. Of course it must read "our"...

File: nx-X11/programs/Xserver/hw/nxagent/Clipboard.c (hunk 3 above)
Comment (uli42) [reply nx-X11/programs/Xserver/hw/nxagent/Clipboard.c:2398]: yes, it does relate to the TimeStamp data type change. I would have needed to change all those printfs where the lastTimeChanged was referred to. The easier way was to replace those printfs by a call to central subroutine that already printed everthing and was already uptodate regarding the TimeStamp change.

File: nx-X11/programs/Xserver/hw/nxagent/Clipboard.c (hunk 4 above)
Comment (uli42) [reply nx-X11/programs/Xserver/hw/nxagent/Clipboard.c]: fixed it otherwise...

**Pull request merged**